wire-ios/Wire-iOS/Sources/UserInterface/Location/AppLocationManager.swift

```diff
@@ -0,0 +1,85 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreLocation
+
+protocol AppLocationManagerDelegate: AnyObject {
+
+    func didUpdateLocations(_ locations: [CLLocation])
+
+    func didFailWithError(_ error: Error)
+
+    func didChangeAuthorization(status: CLAuthorizationStatus)
+
+}
+
+final class AppLocationManager: NSObject {
```

Comment:
I'm wondering what the purpose of `AppLocationManager` is as it appears to wrap `CLLocationManager` without adding anything extra.

Reply:
I actually like it, since it allows for mocking the `CLLocationManager`.
I would have chosen a different name, though.
With `UserNotificationCenterAbstraction` and `UserNotificationCenterWrapper` I did the same.

Reply:
That was my intention. A better separation of concerns @johnxnguyen

Reply:
But doesn't it need to be abstracted with a protocol to allow mocking? Or maybe I'm missing something.

Reply:
As a protocol I introduced `AppLocationManagerDelegate`. Did you have anything else in mind?

```
protocol AppLocationManagerDelegate: AnyObject {

    func didUpdateLocations(_ locations: [CLLocation])

    func didFailWithError(_ error: Error)

    func didChangeAuthorization(status: CLAuthorizationStatus)

}
```

Reply:
I agree, a protocol is missing, I haven't noticed it.

Reply:
Done: https://github.com/wireapp/wire-ios/pull/1377/commits/681fc1f6c82b786cc8ecf862a7c90e67c3c01c03

@johnxnguyen @caldrian